### PR TITLE
BUILD and lock file improvements

### LINKED ISSUE #28: BUILD and lock file improvements
- [x] add `-- --tar` to the BUILD instructions
- [x] Also add a sha and quick opportunity for an up-to-date check (on the repos in file)
- [ ] Also add checker rule

```py
#
# DO NOT EDIT THIS FILE MANUALLY!
#
# To update, call `bazel run @rules_ros//repos/config:repos_lock.update -- --tar` with the right distro set in the WORKSPACE
#
# SHA OF //path/to:repos_index SHA
#
```

### PR DESCRIPTION
**Pull request by michael-sbarra-apexai**
Closes #28

### CODE REVIEW

**Comment by michael-sbarra-apexai** [comment]
@kilian-funk @evan-flynn-apexai Items 1 and 2 from #28 are implemented. Not yet implemented is the checker rule, to ensure the IN file hasn't been modified without an update to the lock file.

File: repos/config/detail/BUILD.bazel [hunk 1]
```diff
@@ -36,3 +36,16 @@ py_binary(
     visibility = ["//visibility:public"],
     deps = [requirement("pyyaml")],
 )
+
+sh_test(
+    name = "test_repos_index_sha256",
+    srcs = ["test/test_repos_index_sha256.sh"],
+    args = [
+        "$(execpath @rules_ros//repos/config:setup_humble.lock.bzl)",  # TODO
```
Comment (michael-sbarra-apexai) [inline repos/config/detail/BUILD.bazel:44]: Almost there, just need to swap this out with the current Workspace lock file.

File: repos/config/setup_dashing.lock.bzl [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #
 # To update, call `bazel run @rules_ros//repos/config:repos_lock.update` with the right distro set in the WORKSPACE
 #
+# SHA256 of external/ros2/ros2.repos: da16d26ba1c05ea20cedf2f01250ed859ee255565053dd9650c831ff80e1f6ea
```
Comment (michael-sbarra-apexai) [inline repos/config/setup_dashing.lock.bzl:6]: This was manually determined. Got the following error:

```
ERROR: No such repository '@eProsima.Fast-DDS': Repository '@eProsima.Fast-DDS' is not defined
ERROR: Skipping '@eProsima.Fast-DDS//...': no targets found beneath ''
```

File: repos/config/setup_eloquent.lock.bzl [hunk 3]
```diff
@@ -3,6 +3,7 @@
 #
 # To update, call `bazel run @rules_ros//repos/config:repos_lock.update` with the right distro set in the WORKSPACE
 #
+# SHA256 of external/ros2/ros2.repos: 71922db335c9a2092fb095e66b1e860f9713854bf704e0e00dc5910f7ec111a8
```
Comment (michael-sbarra-apexai) [inline repos/config/setup_eloquent.lock.bzl:6]: This was manually determined. Got the following error:

```
ERROR: No such repository '@eProsima.Fast-DDS': Repository '@eProsima.Fast-DDS' is not defined
ERROR: Skipping '@eProsima.Fast-DDS//...': no targets found beneath ''
```

File: repos/config/detail/BUILD.bazel (hunk 1 above)
Comment (michael-sbarra-apexai) [reply repos/config/detail/BUILD.bazel:44]:
```suggestion
       "$(execpath {})".format(REPOS_INDEX_FILE),
```

**Comment by michael-sbarra-apexai** [comment]
@evan-flynn-apexai @kilian-funk This MR is ready.

**Review by evan-flynn-apexai — APPROVED** [review]

**Comment by michael-sbarra-apexai** [comment]
@evan-flynn-apexai CI was failing to start because `run` doesn't accept a list. I fixed in the latest push, and added explicit testing of the two packages that are able to be tested in this repo.

**Review by evan-flynn-apexai — APPROVED** [review]

**Pull request merged**